contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1130,4 +1130,108 @@ split_object_name(char *name)
 	return res;
 }
 
+Oid 
+tsql_get_proc_nsp_oid(Oid object_id, Oid user_id)
+{
+	Oid namespace_oid = InvalidOid;
+	HeapTuple tuple;
+	bool isnull;
+
+	/* retrieve namespace oid in pg_proc by oid */
+	tuple = SearchSysCache1(PROCOID, CStringGetDatum(object_id));
+
+	if (HeapTupleIsValid(tuple))
+	{
+		(void) SysCacheGetAttr(PROCOID, tuple,
+								Anum_pg_proc_pronamespace,
+								&isnull);		
+		if(!isnull)
+		{
+			Form_pg_proc proc = (Form_pg_proc) GETSTRUCT(tuple);
+			if(OidIsValid(proc->oid))
+			if (pg_proc_aclcheck(object_id, user_id, ACL_EXECUTE) == ACLCHECK_OK)
+			{
+				namespace_oid = proc->pronamespace;
+			}
+		}
+		ReleaseSysCache(tuple);
+	}
+	return namespace_oid;
+}
+
+Oid 
+tsql_get_constraint_nsp_oid(Oid object_id, Oid user_id){
 
+	Oid namespace_oid = InvalidOid;
+	HeapTuple tuple;
+	bool isnull;
+
+	/* retrieve namespace oid in pg_proc by oid */
+	tuple = SearchSysCache1(CONSTROID, CStringGetDatum(object_id));
+
+	if (HeapTupleIsValid(tuple))
+	{
+		(void) SysCacheGetAttr(CONSTROID, tuple,
+								Anum_pg_constraint_connamespace,
+								&isnull);		
+		if(!isnull)
+		{
+			Form_pg_constraint con = (Form_pg_constraint) GETSTRUCT(tuple);
+			if (OidIsValid(con->oid))
+			{	
+				/* user should have permission of table associated with constraint */
+				if (OidIsValid(con->conrelid))
+				{
+					if(pg_class_aclcheck(con->conrelid, user_id, ACL_SELECT) == ACLCHECK_OK)
+						namespace_oid = con->connamespace;
+				}
+				else
+					namespace_oid = con->connamespace;
```

Comment:
In postgres it is possible to have constraints that are not on tables, but not in TSQL. I have added the else condition to generalise the utility of the helper function, so that it can be used for other constraints too.

Reply:
we should have test case of each if/else condition. If this use case is for non table constraints, add a test case for the same.